token_lobby_bypass/mod_token_lobby_bypass.lua

```diff
@@ -0,0 +1,44 @@
+--- Plugin to allow users to bypass lobby based on attribute in JWT.
+---
+--- This module should be added to the main muc component.
+---
+
+local LOGLEVEL = "debug";
+
+local muc_util = module:require "muc/util";
+local valid_affiliations = muc_util.valid_affiliations;
+local is_healthcheck_room = module:require "util".is_healthcheck_room;
+
+module:log("info", "loaded");
+
+module:hook("muc-occupant-pre-join", function (event)
+    local room, occupant = event.room, event.occupant;
+
+    if is_healthcheck_room(room.jid) then
+        module:log(LOGLEVEL, "skip for room %s occupant %s", room.jid, occupant.bare_jid)
+        return;
+    end
+
+    if not event.origin.auth_token then
+        module:log(LOGLEVEL, "skip user with no token %s", event.origin.from)
+        return
+    end
+
+    local context = event.origin.jitsi_meet_context_features;
+
+    if context then
+        if context['lobby_bypass'] == true then
+            module:log(LOGLEVEL, "Bypassing lobby for room %s occupant %s", room.jid, occupant.bare_jid);
+
+            occupant.role = 'participant';
+
+            -- set affiliation to "member" if not yet assigned by other plugins
+            local affiliation = room:get_affiliation(occupant.bare_jid);
+            if valid_affiliations[affiliation or "none"] < valid_affiliations.member then
+                module:log(LOGLEVEL, "Setting affiliation for %s -> member", occupant.bare_jid);
+                room:set_affiliation(true, occupant.bare_jid, 'member');
```

Comment:
Hum, I wonder is this ok ...when this module is on it changes all participants to be members of rooms even if lobby is not enabled ...

Reply:
Good point. 

In the non-comprehensive testing I've done, the existing affiliation is always `nil` but that just be down to hook priority as you pointed out.

Perhaps it should check `room._data.lobbyroom` and not do anything at all if room has no activated lobby? Or should we do more with the affiliation checks?

Reply:
I would say less operations is better :) especially on single thread for thousands of users :) 
Maybe just add the check for room._data.lobbyroom